### ISSUE TITLE
static IP configured via NM keyfile is not respected on first boot

### ISSUE THREAD

**Issue by miabbott** [coreos/fedora-coreos-tracker#358]
I was testing the NM keyfile in the docs PR (coreos/fedora-coreos-docs#30) and found that the static IP was not respected on first boot.

The keyfile was correctly landed via Igntion as such:

```
$ sudo cat /etc/NetworkManager/system-connections/eth0.nmconnection 
[connection]
type=ethernet
interface-name=eth0

[ethernet]
mac-address=52:54:00:0d:5c:06

[ipv4]
method=manual
addresses=192.168.124.120/24
gateway=192.168.124.1
dns=192.168.124.1;1.1.1.1;8.8.8.8
dns-search=redhat.com
```

But the `eth0` interface did not come up with the static IP, rather it came up with a DHCP assigned address.

I was able to force it to use the static IP after using `nmcli connection down eth0`

Additionally, on subsequent boots, the static IP **does** come up correctly.

This feels like another case where bringing down the networking in the initrd would solve the issue (re:  https://github.com/coreos/fedora-coreos-tracker/issues/356#issuecomment-579466110)

**Comment by ServerNinja** [comment]
I'm experiencing the same issue. I'm inserting an ignition file via pxe to configure an install. Ignition successfully installs fedora core but it grabs an IP from DHCP on first boot. When I do a `sudo reboot`, it comes back up with the correct static IP I defined in the ignition file during the install.

**Comment by cgwalters** [comment]
> This feels like another case where bringing down the networking in the initrd would solve the issue

Yes.  Or [NM in the initrd](https://github.com/coreos/fedora-coreos-config/pull/259).

**Comment by kai-uwe-rommel** [comment]
Is this going to be fixed?
It's a serious issue ...
I'm going to write scripts to create a whole bunch of VMs with static IPs ...

**Comment by jlebon** [comment]
We're working on this in https://github.com/coreos/ignition-dracut/pull/159.

**Comment by dustymabe** [comment]
We are now [using NetworkManager in the initramfs](https://github.com/coreos/fedora-coreos-tracker/issues/394) and also [propagating network information from the initramfs (kargs) when appropriate](https://github.com/coreos/fedora-coreos-tracker/issues/394#issuecomment-599721173), which we think fixes this issue.

See https://github.com/coreos/fedora-coreos-tracker/issues/394#issuecomment-604598128 and the preceding discussion for more details.

**Comment by LorbusChris** [comment]
@miabbott @dustymabe did the fix for this land in 4.4 boot images?

**Comment by cgwalters** [comment]
> @miabbott @dustymabe did the fix for this land in 4.4 boot images?

No, RHCOS 4.4 only special cases ifcfg files right now, see https://github.com/coreos/ignition-dracut/pull/89
Hopefully 4.6 will backport this stuff.
